### PR TITLE
packaging/opensuse: build position-independent binaries

### PR DESCRIPTION
**Pull request by zyga**
This patch adds appropriate flags to build position independent
C-binaries. This fixes the following rpmlint warnings:

[  108s] snapd.x86_64: W: position-independent-executable-suggested /usr/lib/snapd/snap-discard-ns
[  108s] snapd.x86_64: W: position-independent-executable-suggested /usr/lib/snapd/snap-gdb-shim
[  108s] snapd.x86_64: W: position-independent-executable-suggested /usr/lib/systemd/system-generators/snapd-generator
[  108s] This executable should be position independent (all binaries should).  Check
[  108s] that it is built with -fPIE/-fpie in compiler flags and -pie in linker flags.

In addition, since our build system correctly filters out those flags
from statically linked binaries, simplify the code and reword the
comment that sets those flags.

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

### CODE REVIEW

**Comment by zyga** [comment]
The build works on Tumbleweed, investigating on leap now.

File: packaging/opensuse/snapd.spec [hunk 1]
```diff
@@ -134,13 +134,22 @@ the system:snappy repository.
...
-# our build of static binaries for snap-confine. Also see
-# https://bugzilla.redhat.com/show_bug.cgi?id=1343892
+# Enable hardening; Also see https://bugzilla.redhat.com/show_bug.cgi?id=1343892
 CFLAGS="$RPM_OPT_FLAGS -fPIC -Wl,-z,relro -Wl,-z,now"
 CXXFLAGS="$RPM_OPT_FLAGS -fPIC -Wl,-z,relro -Wl,-z,now"
+LDFLAGS=""
+# On openSUSE Leap 15 or more recent build position independent executables.
+# For a helpful guide about the versions and macros used below, please see:
+# https://en.opensuse.org/openSUSE:Build_Service_cross_distribution_howto
+%if (0%{?sle_version} >= 150000 && 0%{?is_opensuse}) || %if 0%{?suse_version} > 1500
```
Comment (Conan-Kudo) [inline packaging/opensuse/snapd.spec]: Just use `0%{?suse_version} >= 1500` here.

File: packaging/opensuse/snapd.spec [hunk 2]
```diff
@@ -134,13 +134,22 @@ the system:snappy repository.
...
+CFLAGS="$CFLAGS -fPIE -fpie"
+CXXFLAGS"$CXXFLAGS -fPIE -pie"
```
Comment (Conan-Kudo) [inline packaging/opensuse/snapd.spec]: You don't set both `-fPIE` and `-fpie`. Only set `-fPIE`.

Comment (zyga) [reply packaging/opensuse/snapd.spec]: Thanks, I'll tweak that,

**Pull request merged**